googlebigtable/src/main/java/com/yahoo/ycsb/db/GoogleBigtableClient.java

```diff
@@ -0,0 +1,443 @@
+import com.google.bigtable.v1.Family;
+import com.google.bigtable.v1.MutateRowRequest;
+import com.google.bigtable.v1.Mutation;
+import com.google.bigtable.v1.ReadRowsRequest;
+import com.google.bigtable.v1.Row;
+import com.google.bigtable.v1.RowFilter;
+import com.google.bigtable.v1.RowRange;
+import com.google.bigtable.v1.Mutation.DeleteFromRow;
+import com.google.bigtable.v1.Mutation.SetCell;
+import com.google.bigtable.v1.RowFilter.Chain.Builder;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.grpc.BigtableDataClient;
+import com.google.cloud.bigtable.grpc.BigtableSession;
+import com.google.cloud.bigtable.grpc.async.AsyncExecutor;
+import com.google.cloud.bigtable.grpc.async.HeapSizeManager;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.util.ByteStringer;
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+
+/**
+ * Google Bigtable Proto client for YCSB framework.
+ * 
+ * Bigtable offers two APIs. These include a native Protobuf GRPC API as well as 
+ * an HBase API wrapper for the GRPC API. This client implements the Protobuf 
+ * API to test the underlying calls wrapped up in the HBase API. To use the 
+ * HBase API, see the hbase10 client binding.
+ */
+public class GoogleBigtableClient extends com.yahoo.ycsb.DB {
+  public static final Charset UTF8_CHARSET = Charset.forName("UTF8");
+  
+  /** Property names for the CLI. */
+  private static final String ASYNC_MUTATOR_MAX_MEMORY = "mutatorMaxMemory";
+  private static final String ASYNC_MAX_INFLIGHT_RPCS = "mutatorMaxInflightRPCs";
+  private static final String CLIENT_SIDE_BUFFERING = "clientbuffering";
+  
+  /** Must be an object for synchronization and tracking running thread counts. */ 
+  private static Integer threadCount = 0;
+  
+  /** This will load the hbase-site.xml config file and/or store CLI options. */
+  private static final Configuration CONFIG = HBaseConfiguration.create();
+  
+  /** Print debug information to standard out. */
+  private boolean debug = false;
+  
+  /** Global Bigtable native API objects. */ 
+  private static BigtableOptions options;
+  private static BigtableSession session;
+  
+  /** Thread loacal Bigtable native API objects. */
+  private BigtableDataClient client;
+  private HeapSizeManager heapSizeManager;
+  private AsyncExecutor asyncExecutor;
+  
+  /** The column family use for the workload. */
+  private byte[] columnFamilyBytes;
+  
+  /** Cache for the last table name/ID to avoid byte conversions. */
+  private String lastTable = "";
+  private byte[] lastTableBytes;
+  
+  /**
+   * If true, buffer mutations on the client. This is the default behavior for
+   * HBaseClient. For measuring insert/update/delete latencies, client side
+   * buffering should be disabled.
+   */
+  private boolean clientSideBuffering = true;
```

Comment:
We should be defaulting to no client side buffering. We should correct any hbase clients doing this.